src/polymer/iron-flex-layout-classes.ts

```diff
@@ -0,0 +1,262 @@
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import {treeAdapters} from 'parse5';
+import {Document, ParsedHtmlDocument, Severity} from 'polymer-analyzer';
+
+import {HtmlRule} from '../html/rule';
+import {registry} from '../registry';
+import {FixableWarning} from '../warning';
+
+import stripIndent = require('strip-indent');
+
+import {elementSelectorToPredicate} from '../html/util';
+
+const p = dom5.predicates;
+
+const styleModules = [
+  {
+    name: 'iron-flex',
+    selector: elementSelectorToPredicate(
+        '.layout.horizontal, .layout.vertical, .layout.inline, .layout.wrap,' +
+        '.layout.no-wrap, .layout.center, .layout.center-center, ' +
+        '.layout.center-justified, .flex, .flex-auto, .flex-none')
+  },
+  {
+    name: 'iron-flex-reverse',
+    selector: elementSelectorToPredicate(
+        '.layout.horizontal-reverse, .layout.vertical-reverse, ' +
+        '.layout.wrap-reverse')
+  },
+  {
+    name: 'iron-flex-alignment',
+    // Skip `.layout.center, .layout.center-center, .layout.center-justified`
+    // as they're already defined in the `iron-flex` module.
+    selector: elementSelectorToPredicate(
+        '.layout.start, .layout.end, .layout.baseline, .layout.start-justified, ' +
+        '.layout.end-justified, .layout.around-justified, .layout.justified, ' +
+        '.self-start, .self-center, .self-end, .self-stretch, .self-baseline, ' +
+        '.layout.start-aligned, .layout.end-aligned, .layout.center-aligned, ' +
+        '.layout.between-aligned, .layout.around-aligned')
+  },
+  {
+    name: 'iron-flex-factors',
+    // Skip `.flex` as it's already defined in the `iron-flex` module.
+    selector: elementSelectorToPredicate(
+        '.flex-1, .flex-2, .flex-3, .flex-4, .flex-5, .flex-6, .flex-7, ' +
+        '.flex-8, .flex-9, .flex-10, .flex-11, .flex-12')
+  },
+  {
+    name: 'iron-positioning',
+    // Skip `[hidden]` as it's a too generic selector.
+    selector: elementSelectorToPredicate(
+        '.block, .invisible, .relative, .fit, body.fullbleed, ' +
+        '.scroll, .fixed-bottom, .fixed-left, .fixed-top, .fixed-right')
+  }
+];
+
+const styleModulesRegex = /iron-(flex|positioning)/;
+
+const isStyleInclude = p.AND(p.hasTagName('style'), p.hasAttr('include'));
+
+class IronFlexLayoutClasses extends HtmlRule {
+  code = 'iron-flex-layout-classes';
+  description = stripIndent(`
+    Warns when iron-flex-layout classes are used without including the style modules.
+  `).trim();
+
+  async checkDocument(parsedDocument: ParsedHtmlDocument, document: Document) {
+    const warnings: FixableWarning[] = [];
+
+    this.convertDeclarations(parsedDocument, document, warnings);
```

Comment:
I'd inline `convertDeclarations`. I did that in `content-to-slot-declarations` back when the usages and declarations passes were in a single rule rather than split out into two.

Reply:
Done